src/components/frame/Extensions/pages/MultiSigPage/hooks.ts

```diff
@@ -0,0 +1,161 @@
+  type MultiSigSetupFormValues,
+} from './types.ts';
+import {
+  getDomainThresholds,
+  getInitialDomainConfig,
+  getInitialGlobalThreshold,
+  getThresholdType,
+} from './utils.ts';
+
+export const useBackNavigation = ({ extensionData }) => {
+  const navigate = useNavigate();
+
+  /*
+   * If we arrive here but the extension is not installed go back to main extension details page
+   */
+  useEffect(() => {
+    if (!isInstalledExtensionData(extensionData)) {
+      navigate(-1);
+    }
+  }, [extensionData, navigate]);
+};
+
+export const useThresholdData = ({ extensionData }) => {
+  const multiSigConfig = extensionData.params?.multiSig || null;
+
+  const {
+    colony: { colonyAddress, domains },
+  } = useColonyContext();
+
+  const {
+    register,
+    setValue,
+    getValues,
+    trigger,
+    formState: { errors, isValid },
+    reset,
+  } = useForm<MultiSigSetupFormValues>({
+    defaultValues: {
+      globalThreshold: getInitialGlobalThreshold(multiSigConfig),
+    },
+    mode: 'onChange',
+  });
+
+  const { handleIsVisible, handleSetActionType, resetAll, callback } =
+    useContext(ExtensionSaveSettingsContext);
+
+  const [thresholdType, setThresholdType] = useState<MultiSigThresholdType>(
+    getThresholdType(multiSigConfig?.colonyThreshold),
+  );
+
+  const [domainThresholdConfigs, setDomainThresholdConfigs] = useState<
+    DomainThresholdConfig[]
+  >([]);
+
+  useEffect(() => {
+    handleIsVisible(true);
+    handleSetActionType(ActionTypes.MULTISIG_SET_THRESHOLDS);
+
+    return () => {
+      resetAll();
+      reset();
+    };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  useEffect(() => {
+    setValue('colonyAddress', colonyAddress);
+  }, [colonyAddress, setValue]);
+
+  useEffect(() => {
+    if (!domains || !domains.items || !multiSigConfig) {
+      return;
+    }
+
+    const domainsExcludingRoot = domains.items.filter(
+      (domain): domain is NonNullable<typeof domain> =>
+        domain !== null && !domain.isRoot,
+    );
+
+    const tempDomainThresholdConfigs = domainsExcludingRoot.map((domain) => {
+      const { threshold, ...rest } = getInitialDomainConfig(
+        domain,
+        multiSigConfig,
+      );
+
+      setValue(rest.name, threshold);
+
+      return rest;
+    });
+
+    setDomainThresholdConfigs(tempDomainThresholdConfigs);
+  }, [domains, multiSigConfig, setValue]);
+
+  useImperativeHandle(callback, () => ({
+    getValues: async () => {
+      await trigger(undefined, { shouldFocus: true });
+      if (!isValid) throw new Error('Error within form');
+
+      const values = getValues();
+
+      return {
+        colonyAddress,
+        globalThreshold:
+          thresholdType === MultiSigThresholdType.MAJORITY_APPROVAL
+            ? 0
+            : values.globalThreshold,
+        domainThresholds: getDomainThresholds(
+          values,
+          domainThresholdConfigs,
+          thresholdType,
+        ),
+      };
+    },
+  }));
```

Comment:
Deps would be ideal for this hook, otherwise it will just keep on running on every render, like useEffect:

```js
useImperativeHandle(ref, () => {
  return {
    // boogie dance
  };
}, []); <<-- deps
```